### PR TITLE
Add parallel implementation of operations on Evaluations

### PR DESCRIPTION
**Pull request by ValarDragon**
## Description

Add parallel implementation of operations on the Evaluations struct. 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (main)
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [ ] Wrote unit tests - should be covered by existing tests? Not sure if there are any though. Batch inversion change is covered by prior tests.
- [x] Updated relevant documentation in the code - n/a
- [x] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

File: poly/src/evaluations.rs [hunk 1]
```diff
@@ -146,8 +146,7 @@ impl<'a, F: FftField, D: EvaluationDomain<F>> DivAssign<&'a Evaluations<F, D>>
     #[inline]
     fn div_assign(&mut self, other: &'a Evaluations<F, D>) {
         assert_eq!(self.domain, other.domain, "domains are unequal");
-        self.evals
-            .iter_mut()
+        ark_std::cfg_iter_mut!(self.evals)
```
Comment (Pratyush) [inline poly/src/evaluations.rs:149]: I guess we should use batch inversion here? (I know the old code didn’t)

Comment (ValarDragon) [reply poly/src/evaluations.rs:149]: Sure, I'll add a parallel batch inversion method.

**Comment by Pratyush** [comment]
hmm the new batch inversion does more inversions, no? maybe you can improve the cost further by doing a `serial_batch_inversion` across all the per-batch inversions?

**Comment by ValarDragon** [comment]
The new batch inversion only does one inversion per thread, which is all done in parallel

**Review by Pratyush — APPROVED** [review]

**Pull request merged**